internal/server/iigointernal/orchestration.go

```diff
@@ -68,48 +68,48 @@ var presidentPointer roles.President = nil
 // iigoClients holds pointers to all the clients
 var iigoClients map[shared.ClientID]baseclient.Client
 
-var iigoRoleStates gamestate.IIGOBaseRoles
-
 // RunIIGO runs all iigo function in sequence
 func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.Client) error {
 
 	// TODO: Get Client pointers from gamestate https://imgur.com/a/HjVZIkh
 	iigoClients = *clientMap
-	iigoRoleStates = g.IIGOInfo
 
 	// Initialise IDs
-	featureJudge.ID = JudgeIDGlobal
-	featureSpeaker.Id = SpeakerIDGlobal
-	featurePresident.ID = PresidentIDGlobal
+	judicialBranch.JudgeID = JudgeIDGlobal
+	legislativeBranch.SpeakerID = SpeakerIDGlobal
+	executiveBranch.ID = PresidentIDGlobal
 
 	// Initialise iigointernal with their clientVersions
-	featureJudge.clientJudge = judgePointer
-	featurePresident.clientPresident = presidentPointer
-	featureSpeaker.clientSpeaker = speakerPointer
+	judicialBranch.clientJudge = judgePointer
+	executiveBranch.clientPresident = presidentPointer
+	legislativeBranch.clientSpeaker = speakerPointer
 
 	// Withdraw the salaries
-	errWithdrawPresident := featureJudge.withdrawPresidentSalary(g)
-	errWithdrawJudge := featureSpeaker.withdrawJudgeSalary(g)
-	errWithdrawSpeaker := featurePresident.withdrawSpeakerSalary(g)
+	errWithdrawPresident := judicialBranch.withdrawPresidentSalary(g)
+	errWithdrawJudge := legislativeBranch.withdrawJudgeSalary(g)
+	errWithdrawSpeaker := executiveBranch.withdrawSpeakerSalary(g)
```

Comment:
Here on the initial turn wont these be nil pointer?

Reply:
do you mean the judicialbranch etc? we initialise these globally on lines 17-37

Reply:
we have now randomly assigned roles for the first turn - thanks!

Reply:
well, not super randomly - islands 1, 2 and 3.